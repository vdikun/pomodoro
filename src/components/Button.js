import React, { Component } from 'react';
import {
  TouchableOpacity,
  Text,
  View,
} from 'react-native';
import styles from './../styles/common';

export default Button = ({ children, onPress }) => {
    const { buttonStyle, buttonTextStyle } = styles;
    return (
        <TouchableOpacity style={ buttonStyle } onPress={ onPress }>
            <Text style={ buttonTextStyle }>
                {children}
            </Text>
        </TouchableOpacity>
    );
};