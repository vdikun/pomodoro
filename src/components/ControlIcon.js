import React from 'react';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import styles, { Black } from './../styles/components/ControlIcon.style';

style = {
    opacity: 0.7,
    margin: 5
}

const ControlIcon = ({ iconName, onPress }) => {
    return (
        <TouchableOpacity onPress={() => onPress()} style={styles.view}>
            <Icon name={iconName} size={40} color={Black} />
        </TouchableOpacity>
    );
};

export default ControlIcon;