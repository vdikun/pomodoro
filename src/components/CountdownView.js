import React, { Component } from 'react';
import {
  Text,
} from 'react-native';
import { connect } from 'react-redux';
import { tick } from './../actions';
import Timer from './Timer';
import styles from './../styles/components/Timer.style';

class CountdownView extends Component {

	startTimer() {
		if (!this._interval) {
			this._interval = setInterval(() => {
				this.props.tock()
			}, 1000);
		}
	}

	stopTimer() {
		if (this._interval) {
			clearInterval(this._interval);
			this._interval = null;
		}
	}

	render() {
		const { activeTimer, secondsRemaining } = this.props;
		if (activeTimer) {
			this.startTimer();
    	const { timerStyle } = styles;
    	return (
	    	<Text style={ timerStyle }>
	        	{ Timer.showTimerText(secondsRemaining) }
	        </Text>
    	);
		} else {
			this.stopTimer();
			return null;
		}
	}
};

const mapStateToProps = (state) => {
  return {
		secondsRemaining: state.pomodoro.secondsRemaining,
		activeTimer: state.pomodoro.activeTimer
  }
};

const mapDispatchToProps = (dispatch) => {
	return { 
			tock: () => {
					dispatch(tick());
			}
	};
}

export default connect(mapStateToProps, mapDispatchToProps)(CountdownView);