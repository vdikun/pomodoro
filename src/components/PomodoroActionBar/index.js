import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import StartControl from '../controls/StartControl';
import ToggleControl from '../controls/ToggleControl';
import PauseControl from '../controls/PauseControl';
import pomodoroState from '../../reducers/state';
import styles from './styles';

const PomodoroActionBar = ({ theState }) => {

    if (theState == pomodoroState.DISABLED) {
        return (
            <View style={styles.controlStyle}>
                <StartControl />
            </View>
        );
    } else {
        return (
            <View style={styles.controlStyle}>
                <StartControl />
                <ToggleControl />
                <PauseControl />
            </View>
        );
    }
};

const mapStateToProps = (state) => {
  return {
    theState: state.pomodoro.pomodoroState
  }
};

export default connect(mapStateToProps)(PomodoroActionBar);