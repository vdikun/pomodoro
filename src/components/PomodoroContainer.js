import React from 'react';
import { connect } from 'react-redux';
import {
  View,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import styles, { Colors } from './../styles/common';

export const PomodoroContainer = ({ children, isPomodoro }) => {
    const { redContainer, greenContainer } = styles;
    var containerStyle = (isPomodoro) ? redContainer : greenContainer;
    var colors = (isPomodoro) ?
    	[ Colors.redMajor, Colors.redMinor ] :
    	[ Colors.greenMajor, Colors.greenMinor ];
    return (
        <LinearGradient colors={colors} style={ containerStyle }>
            {children}
        </LinearGradient>
    );
};

const mapStateToProps = (state) => {
  return {
  	isPomodoro: state.pomodoro.isPomodoro
  }
};

export default connect(mapStateToProps)(PomodoroContainer);