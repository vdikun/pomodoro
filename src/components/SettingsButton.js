import React from 'react';
import {
  Image,
  TouchableOpacity
} from 'react-native';
import styles from './../styles/components/SettingsButton.style';

const settingsIcon = './../../img/settings-18dp.png';

const SettingsButton = ({ onPress }) => {
    return (
        <TouchableOpacity
        style={ styles.settingsButton }
        onPress={ () => { onPress() } }>
            <Image 
              style={ styles.settingsButton }
              source={require(settingsIcon)} />
        </TouchableOpacity>
    );
};

export default SettingsButton;