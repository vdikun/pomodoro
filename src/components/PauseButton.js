import React from 'react';
import { connect } from 'react-redux';
import { toggleTimer } from './../actions';
import Button from './Button';

const PauseButton = ({ isTimerActive, onPress }) => {
    let buttonText = (isTimerActive) ? "Pause Timer" : "Start Timer";
    return (
        <Button text={buttonText} onPress={() => onPress()} />
    );
};

const mapStateToProps = (state) => {
    return {
      isTimerActive: state.pomodoro.activeTimer,
    }
  };
  
  const mapDispatchToProps = (dispatch) => {
      return { 
          onPress: () => {
              dispatch(toggleTimer());
          }
      }
  };

export default connect(mapStateToProps, mapDispatchToProps)(PauseButton);