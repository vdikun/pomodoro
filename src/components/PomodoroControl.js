import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import StartButton from './StartButton';
import ToggleButton from './ToggleButton';
import PauseButton from './PauseButton';
import pomodoroState from './../reducers/state';
import styles from './../styles/components/PomodoroControl.style';

const PomodoroControl = ({ theState }) => {

    if (theState == pomodoroState.DISABLED) {
        return (
            <View style={styles.controlStyle}>
                <StartButton />
            </View>
        );
    } else {
        return (
            <View style={styles.controlStyle}>
                <StartButton />
                <ToggleButton />
                <PauseButton />
            </View>
        );
    }
};

const mapStateToProps = (state) => {
  return {
    theState: state.pomodoro.pomodoroState
  }
};

export default connect(mapStateToProps)(PomodoroControl);