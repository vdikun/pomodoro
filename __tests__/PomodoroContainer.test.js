import React from 'react';
import { PomodoroContainer } from './../src/components/PomodoroContainer';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<PomodoroContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});