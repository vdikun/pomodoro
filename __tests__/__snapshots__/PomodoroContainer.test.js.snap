// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4292540326,
      4287353373,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DAF7A6",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
